assets/data-port/stepper/index.jsx

```diff
@@ -0,0 +1,102 @@
+import classnames from 'classnames';
+
+/**
+ * Helper method to update the steps when moving to the next one.
+ *
+ * @param  {Array} steps  The current steps.
+ * @return {Array}        The steps after moving.
+ */
+const moveToNext = ( steps ) => {
+	const newSteps = [ ...steps ];
+
+	for ( let i = 0; i < newSteps.length; i++ ) {
+		if ( newSteps[ i ].isActive ) {
+			newSteps[ i ].isComplete = true;
+
+			if ( i + 1 < newSteps.length ) {
+				newSteps[ i + 1 ].isActive = true;
+				newSteps[ i ].isActive = false;
+			}
+
+			return newSteps;
+		}
+	}
+
+	throw new Error( 'No active step.' );
+};
+
+/**
+ * Helper method to update the steps when the active step is completed.
+ *
+ * @param  {Array} steps  The current steps.
+ * @return {Array}        The steps after completing the current one.
+ */
+const completeCurrentStep = ( steps ) => {
+	const newSteps = steps.map( ( step ) => {
```

Comment:
@renatho [said](https://github.com/Automattic/sensei/pull/3180#discussion_r428659674):

Here the same about return the complete object. Here is simpler because we can use the map.

```javascript
const completeCurrentStep = ( steps ) =>
  steps.map( ( step ) => ( {
    ...step,
    isComplete: true,
  } ) );
```

Reply:
Fixed 1764f291c9dd4a5f1b5c5d9cb816fedd47207b6c

Reply:
Oops! Here I mean about the mutating.